VMT/templates/KCSA.md

```diff
@@ -0,0 +1,71 @@
+# Kata Containers security advisories (KCSA) template
+
+> **FIXME** This template most likely needs review and rewrite.
+
+The document is first submitted as a markdown formatted description to the KCSA project using this template:
+
+```
+date: YYYY-MM-DD
+
+id: KCSA-$NUM
+
+title: '$TITLE'
+
+description: '$DESCRIPTION_CONTENT'
+
+affected-products:
+
+  - product: $PROJECT
+    version: $AFFECTED_VERSIONS
+
+vulnerabilities:
+
+  - cve-id: $CVE
+
+reporters:
+
+  - name: '$CREDIT'
+    affiliation: $CREDIT_AFFILIATION
+    reported:
+      - $CVE
+
+issues:
+
+  links:
+    - https://github.com/kata-containers/$PROJECT/issues/$ISSUE_NR
+
+ reviews:
```

Comment:
Ah, right - that'll be because these templates were derived from the OS YAML ones ;-) - but, as we are an MD based project, I'll clean those up to be valid MD. thx for noticing.

They are referenced throughout the main VMT.md file via hotlinks, hence why they are not overly self-describing by themselves. I thought I had (but looks like I did not) add a README.md in the templates dir to give brief descriptions of their use cases - let me do that...

Reply:
The reason for the OpenStack VMT standardizing on YAML had nothing to do with RST vs MD, and was more about relying on structured data to be able to more consistently mine historical vulnerability data by avoiding freeform text anywhere except where actual prose was required. In our case, the implementation was via a Sphinx extension which merged the structured data with a template in a markup language which could double both as a text-based version to send to public mailing lists and also as intermediate source for HTML rendering in our security advisories site publishing automation.

Reply:
@fungi Sure. Kata does not have an automated method of posting at present. Unless we go that route, I suggest we stick with MD for the project. It's a bit of a choice between simplicity (MD), or potential pre-emptive optimisation for the future (YAML) I think.
I'm happy to be swayed to YAML if the majority feel that is a better path forwards, in which case I'll convert the templates to real YAML (and probably look at `pandoc` to see how its yaml->github-md conversion is, presuming it has one).

Reply:
tbh, given that the docs are almost YAML (I assumed the templates where supposed to be and this indent was just an oversight), I'd vote we start with YAML for machine readability. The CI already has `yamllint` so we can add extra validation into https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh by just expanding https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh#L243.

But wdyt @bergwolf, @egernst, @sboeuf?

Related to my crazy idea: https://github.com/kata-containers/runtime/pull/402#issuecomment-397561226.

Reply:
Let's also not forget that YAML is:

- already used in the project (for example the [versions database](https://github.com/kata-containers/runtime/blob/master/versions.yaml).
- machine readable
- human-readable
- prettier than any other text format (CSV, TOML, JSON)
- supports comments and structured data
- basically :sparkle: pure awesome :sparkle: 
- **standardised** (unlike markdown which has myriad versions). We use [gfm](https://github.github.com/gfm/), but that might be an issue if we were ever to switch code hosting sites. YAML protects us from that and buys us a lot more besides imho ;)

Reply:
**step away** from the coffee @jodh-intel:-)

Reply:
The talk of "converting" YAML to MD seems to be missing the point, I think. As @jodh-intel points out and as its acronym expansion indicates, "YAML Ain't Markup Language" like RST or MD it's a structured data serialization more akin to XML, CSV, TOML or JSON (there are quite a few YAML/JSON translator tools out there, for example). Embedding data from a YAML stream into a document would likely involve a template of some sort implementing the markup and associated boilerplate content.

Reply:
OK. I think we are making this choice look a bit harder than it is. I think we only have two choices:
1) Use YAML, and have some method to convert that YAML to MD to upload to our GitHub docs (something similar to how OS/OSSA does it [here](https://github.com/openstack/ossa) I believe?
  This gains us a lot of uniformity and automation and checking, but at the cost of some tooling and complexity.
1) Use MD directly, which makes the process much easier (copy/edit the template, and submit to github), but skips all the automated checking.

I'm not unhappy with (1), it just costs us more work and complexity to get up and running.
Given the only two reviewers so far are favoring (1), if we are happy with the extra burden (and tbh, that is likely to fall on me initially...), then towards YAML we may move.

everybody else - you get a couple of days to chip in here if you want...

Reply:
Just to be clear, I'm not necessarily advocating for any particular solution, merely attempting to provide historical perspective for the choices the OpenStack VMT made in the process from which you're copying some of this. Ultimately I hope that Kata's vulnerability managers will pick a solution which is sane and useful for their situation, but if that includes repeating some of the OpenStack VMT's choices then I'm happy you've found those useful.

And yes, the actual implementation for our YAML integration happens in a Sphinx extension at https://git.openstack.org/cgit/openstack/ossa/tree/doc/source/_exts/vmt.py using a "rst.jinja" Jinja2 template found in the same directory. It's all available under the same Creative Commons Attribution 3.0 Unported License as our content and you're welcome to reuse any of it you find useful.